populate-gems-cache/action.yml

```diff
@@ -0,0 +1,34 @@
+name: Populate Bundler Gems Cache
+description: Populates the Homebrew Bundler RubyGems Cache
+runs:
+  using: "composite"
+  steps:
+    - name: Set up Homebrew
+      id: set-up-homebrew
+      uses: Homebrew/actions/setup-homebrew@master
+
+    - name: Set up RubyGems cache
+      shell: bash
+      run: |
+        cache_key_prefix="$RUNNER_OS"
+        if [[ "$RUNNER_OS" = macOS ]]
+        then
+          macos_full_version="$(sw_vers -productVersion)"
+          macos_version="${macos_full_version%%.*}"
+          arch="$(uname -m)"
+          cache_key_prefix="${macos_version}-${arch}"
+        fi
```

Comment:
Feels like this logic is probably homebrew-core specific? Everywhere else can just use the (GitHub provided) runner name, right?